Flaky test for canary delete

/kind bug

**What steps did you take and what happened:**
```
?       github.com/kubeflow/kfserving/pkg/constants     [no test files]
?       github.com/kubeflow/kfserving/pkg/controller    [no test files]
--- FAIL: TestCanaryDelete (20.57s)
   controller_test.go:748:
       Timed out after 20.000s.
       Expected
           <bool>: false
       to be true
FAIL
```


**What did you expect to happen:**


**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]


**Environment:**

- Istio Version:
- Knative Version:
- KFServing Version:
- Kubeflow version:
- Minikube version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

--- a/pkg/controller/inferenceservice/controller_test.go
+++ b/pkg/controller/inferenceservice/controller_test.go
@@ -733,7 +733,18 @@ func TestCanaryDelete(t *testing.T) {
 	canaryUpdate.Spec.Canary = nil
 	canaryUpdate.Spec.CanaryTrafficPercent = 0
 	g.Expect(c.Update(context.TODO(), canaryUpdate)).NotTo(gomega.HaveOccurred())
-
+	g.Expect(err).NotTo(gomega.HaveOccurred())
+	g.Eventually(requests, timeout).Should(gomega.Receive(gomega.Equal(expectedCanaryRequest)))
+	// Need to wait for update propagate back to controller before checking
+	canaryDelete := &kfserving.InferenceService{}
+	g.Eventually(func() bool {
+		if err := c.Get(context.TODO(), canaryServiceKey, canaryDelete); err != nil {
+			return false
+		}
+		return canaryDelete.Spec.Canary == nil
+	}, timeout).Should(gomega.BeTrue())
+	// Trigger another reconcile
+	g.Expect(c.Update(context.TODO(), canaryDelete)).NotTo(gomega.HaveOccurred())
 	g.Expect(err).NotTo(gomega.HaveOccurred())
 	g.Eventually(requests, timeout).Should(gomega.Receive(gomega.Equal(expectedCanaryRequest)))
 

